src/main/java/com/denizenscript/depenizen/bukkit/objects/jobs/JobsJobTag.java

```diff
@@ -143,93 +151,169 @@ public String toString() {
-                // -->
-                return new ElementTag(jobProgression.getExperience()).getObjectAttribute(attribute);
+    @Override
+    public void adjust(Mechanism mechanism) {
+        // <--[mechanism]
+        // @object JobsJobTag
+        // @name xp
+        // @input ElementTag(Double)
+        // @plugin Depenizen, Jobs
+        // @description
+        // Set the amount of job XP the player has in this job
+        // -->
+        if (mechanism.matches("xp") && mechanism.requireDouble()) {
+            if (jobProgression == null) {
+                mechanism.echoError("This mechanism requires the object to be linked to a player");
+                return;
             }
+            jobProgression.setExperience(mechanism.getValue().asDouble());
         }
 
+        CoreUtilities.autoPropertyMechanism(this, mechanism);
+    }
+
+    public static void registerTags() {
+        PropertyParser.registerPropertyTagHandlers(JobsJobTag.class, tagProcessor);
         // <--[tag]
         // @attribute <JobsJobTag.description>
         // @returns ElementTag
         // @plugin Depenizen, Jobs
+        // @deprecated Use 'JobsJobTag.full_description'
         // @description
         // Returns the description of the job.
+        // Single line description is deprecated in Jobs. Use <@link tag JobsJobTag.full_description> instead.
         // -->
-        else if (attribute.startsWith("description")) {
-            return new ElementTag(job.getDescription()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(ElementTag.class, "description", (attribute, object) -> {
+            return new ElementTag(object.getJob().getDescription());
+        });
 
-        else if (attribute.startsWith("name")) {
-            attribute = attribute.fulfill(1);
-
-            // <--[tag]
-            // @attribute <JobsJobTag.name.short>
-            // @returns ElementTag
-            // @plugin Depenizen, Jobs
-            // @description
-            // Returns the shortened name of the job.
-            // -->
-            if (attribute.startsWith("short")) {
-                return new ElementTag(job.getShortName())
-                        .getObjectAttribute(attribute.fulfill(1));
+        // <--[tag]
+        // @attribute <JobsJobTag.full_description>
+        // @returns ElementTag
+        // @plugin Depenizen, Jobs
+        // @description
+        // Returns the full description of the job.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "full_description", (attribute, object) -> {
+            return new ListTag(object.getJob().getFullDescription());
+        });
+
+        // <--[tag]
+        // @attribute <JobsJobTag.name.short>
+        // @returns ElementTag
+        // @plugin Depenizen, Jobs
+        // @deprecated Use 'JobsJobTag.short_name'
+        // @description
+        // Returns the shortened name of the job.
```

Comment:
This descriptive line shouldn't be in a deprecated tag, only the deprecation notice